src/Application.php

```diff
@@ -49,41 +49,55 @@ public function migrateTransformationConfig(array $transformationConfig): array
             );
 
             if (!isset($transformationsV2[$transformationKey])) {
-                $transformationsV2[$transformationKey] = new TransformationV2(
-                    $transformationConfig['name'],
-                    $row['configuration']['type'],
-                    $row['configuration']['backend'],
-                    (int) $row['configuration']['phase']
-                );
-                if (!empty($transformationConfig['description'])) {
-                    $transformationsV2[$transformationKey]->addDescription($transformationConfig['description']);
+                $transformationV2 = TransformationV2::createFromConfig($transformationConfig, $row);
+                $transformationsV2[$transformationKey][] = $transformationV2;
+            } else {
+                $transformationV2 = null;
+                foreach ($transformationsV2[$transformationKey] as $key => $savedTransformationV2) {
+                    if ($this->isSuitableTransformation($row, $savedTransformationV2) === true) {
+                        $transformationV2 = $savedTransformationV2;
+                        $transformationsV2[$transformationKey][$key] = $transformationV2;
+                        break;
+                    }
```

Comment:
So, here, if the `$row` is suitable for migration then it is discarded?  it looks like nothing is being used from the `row` here?